sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -8,21 +8,25 @@
 import six
 
 if TYPE_CHECKING:
-    from typing import Any, NamedTuple
+    from typing import Any, Optional, NamedTuple
     from typing_extensions import Protocol
 
     AccessToken = NamedTuple("AccessToken", [("token", str), ("expires_on", int)])
 
     class TokenCredential(Protocol):
-        """Protocol for classes able to provide OAuth tokens.
+        """Protocol for classes able to provide OAuth tokens."""
 
-        :param str scopes: Lets you specify the type of access needed.
-        """
+        def get_token(
+            self, *scopes: str, claims: Optional[str] = None, tenant_id: Optional[str] = None, **kwargs: Any
```

Comment:
not 
```py
self, *scopes: str, *, claims: Optional[str] = None, tenant_id: Optional[str] = None, **kwargs: Any
```
?

Reply:
Laurent and I also just learned about this! The `*` argument isn't allowed after `*args` (or `*scopes`, in this case) because `*args` will capture any non-keyword arguments, making `*` unnecessary.